crates/telio-nurse/src/heartbeat.rs

```diff
@@ -262,6 +264,21 @@ impl Runtime for Analytics {
     type Err = ();
 
     async fn wait(&mut self) -> WaitResponse<'_, Self::Err> {
+        let derp_event_future: Pin<Box<dyn Future<Output = Result<Box<Server>, _>> + Send>> =
```

Comment:
futures crate has [BoxFuture](https://docs.rs/futures/latest/futures/future/type.BoxFuture.html) for this.
And also .boxed() method to nicely convert to it, in FutureExt. Use it to remove some boilder plate

Reply:
Also you can use [Either](https://docs.rs/futures/latest/futures/future/enum.Either.html) to avoid heap allocation entirely. and use [left/right_future extension methods](https://docs.rs/futures/latest/futures/future/trait.FutureExt.html#method.left_future)